### PR TITLE
[Feature] Implement HiveStatisticsProvider

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/11349
## Problem Summary(Required) ：

Provide a statistics provider to calculator hive or hudi table statistics

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 179 / 248 (72.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveStatisticsProvider.java|177|246|71.95%|[68, 69, 117, 118, 172, 173, 182, 183, 184, 186, 187, 188, 190, 225, 250, 251, 252, 253, 255, 293, 323, 337, 349, 350, 351, 352, 353, 355, 356, 359, 360, 361, 362, 364, 366, 417, 420, 422, 427, 430, 432, 440, 443, 459, 460, 461, 462, 463, 464, 465, 466, 467, 469, 471, 472, 473, 474, 475, 478, 479, 480, 481, 483, 484, 485, 486, 487, 488, 490]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 204 / 248 (82.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveStatisticsProvider.java|202|246|82.11%|[68, 69, 117, 118, 172, 173, 182, 183, 184, 186, 187, 188, 190, 225, 250, 251, 252, 253, 255, 293, 323, 337, 349, 350, 351, 352, 353, 355, 356, 359, 360, 361, 362, 364, 366, 417, 420, 422, 427, 430, 432, 440, 443, 465]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveStatisticsProvider.java [hunk 1]
```diff
@@ -0,0 +1,492 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveStatisticsProvider.java:1]: 2022-present

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -246,6 +246,7 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     public static final String ALLOW_DEFAULT_PARTITION = "allow_default_partition";
 
     public static final String ENABLE_HIVE_COLUMN_STATS = "enable_hive_column_stats";
+    public static final String HIVE_PARTITION_STATS_SAMPLE_SIZE = "3000";
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:249]: add comment

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveStatisticsProvider.java [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+            return 1;
+        }
+        return fraction;
+    }
+
+    public int getSamplePartitionSize(OptimizerContext session) {
+        return session.getSessionVariable().getHivePartitionStatsSampleSize();
+    }
+
+    static List<String> getPartitionsSample(List<String> partitions, int sampleSize) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveStatisticsProvider.java:452]: add some comment to describe how to sample the partition

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:253]: done

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveStatisticsProvider.java (hunk 3 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveStatisticsProvider.java:455]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 204 / 248 (82.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveStatisticsProvider.java|202|246|82.11%|[68, 69, 117, 118, 172, 173, 182, 183, 184, 186, 187, 188, 190, 225, 250, 251, 252, 253, 255, 293, 323, 337, 349, 350, 351, 352, 353, 355, 356, 359, 360, 361, 362, 364, 366, 417, 420, 422, 427, 430, 432, 440, 443, 468]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**